plugins/intrinsics/src/main/java/org/qbicc/plugin/intrinsics/core/CoreIntrinsics.java

```diff
@@ -368,12 +370,106 @@ public static void registerJavaLangThreadIntrinsics(CompilationContext ctxt) {
+
+        /* private native void start0(); */
+        InstanceIntrinsic start0 = (builder, instance, target, arguments) -> {
+            MethodDescriptor JLT_start0Desc = MethodDescriptor.synthesize(classContext, BaseTypeDescriptor.V, List.of(voidUnaryfunctionPtrDesc, voidPtrDesc));
+
+            /* set java.lang.Thread.threadStatus to runnable and alive */
+            ValueHandle threadStatusHandle = builder.instanceFieldOf(builder.referenceHandle(instance), jltDesc, "threadStatus", BaseTypeDescriptor.I);
+            builder.store(threadStatusHandle, ctxt.getLiteralFactory().literalOf(threadRunnable | threadAlive), MemoryAtomicityMode.NONE);
+
+            /* pass threadWrapper as function_ptr - TODO this will eventually be replaced by a call to CNative.addr_of_function */
+            MethodElement threadWrapperEE = ctxt.getVMHelperMethod("threadWrapper");
+            Literal functionParamLiteral = ctxt.getLiteralFactory().functionLiteralOf(threadWrapperEE.getName(), threadWrapperEE.getType());
+
+            /* call threadWrapper with null parameter so it does nothing - TODO this is a workaround to create a declares statement for threadWrapper in java.lang.Thread */
+            ValueType paramType = threadWrapperEE.getParameters().get(0).getType();
+            MethodDescriptor threadWrapperDesc = MethodDescriptor.synthesize(classContext, voidPtrDesc, List.of(voidPtrDesc));
+            ValueHandle threadWrapperValueHandle = builder.staticMethod(vmHelpersDesc, "threadWrapper", threadWrapperDesc);
+            builder.call(threadWrapperValueHandle, List.of(ctxt.getLiteralFactory().zeroInitializerLiteralOfType(paramType)));
+
+            /* pass java.lang.Thread object as ptr<void> */
+            Value threadVoidPtr = builder.bitCast(instance, (WordType)paramType);
+
+            /* start pthread in VMHelpers */
+            builder.call(builder.staticMethod(vmHelpersDesc, "JLT_start0", JLT_start0Desc), List.of(functionParamLiteral, threadVoidPtr));
+
+            return ctxt.getLiteralFactory().zeroInitializerLiteralOfType(ctxt.getTypeSystem().getVoidType());
+        };
+        intrinsics.registerIntrinsic(jltDesc, "start0", voidDesc, start0);
+
+        /* public final native boolean isAlive(); */
+        InstanceIntrinsic isAlive = (builder, instance, target, arguments) -> {
+            ValueHandle threadStatusHandle = builder.instanceFieldOf(builder.referenceHandle(instance), jltDesc, "threadStatus", BaseTypeDescriptor.I);
+            Value threadStatus = builder.load(threadStatusHandle, MemoryAtomicityMode.NONE);
+            Value aliveState = ctxt.getLiteralFactory().literalOf(threadAlive);
+            Value isThreadAlive = builder.and(threadStatus, aliveState);
+            return builder.isEq(isThreadAlive, aliveState);
+        };
+        intrinsics.registerIntrinsic(jltDesc, "isAlive", booleanDesc, isAlive);
```

Comment:
Again this is OK for now, but the basic problem with using intrinsics for methods that are public like this are that the intrinsic doesn't populate the method's body. This means that if this method were called reflectively we'd have a problem.

It doesn't need to change for now but expect that we'll have to go back and change this to be an actual method at some point.

Reply:
As well as its annoying to debug 😆 as a future pr it would be helpful (to me at least) just to have even a Thread$_native method that just calls this intrinsic